### PR TITLE
Added typing to `export_to_phy`

### PR DESCRIPTION
**Pull request by DradeAW**
Fixes annoying warning from my IDE telling me that `output_folder` needs to be a `str`, not `Path`

### CODE REVIEW

**Comment by DradeAW** [comment]
Btw, `npt.DTypeLike` is really nice to use, but was introduced in `numpy 1.20` (https://numpy.org/doc/stable/reference/typing.html#numpy.typing.DTypeLike)

There is no minimal version specified in `pyproject.toml`
Is this an issue?

**Comment by alejoe91** [comment]
> Btw, `npt.DTypeLike` is really nice to use, but was introduced in `numpy 1.20` (https://numpy.org/doc/stable/reference/typing.html#numpy.typing.DTypeLike)
> 
> There is no minimal version specified in `pyproject.toml` Is this an issue?
> 
> For reference: Numpy 1.20 was released January 30th 2021

We can assume Numpy >= 1.20 ;)

**Comment by DradeAW** [comment]
Perfect ! ☺️

File: src/spikeinterface/exporters/to_phy.py [hunk 1]
```diff
@@ -15,17 +17,17 @@
...
-    compute_amplitudes=True,
-    sparsity=None,
-    copy_binary=True,
-    remove_if_exists=False,
-    peak_sign="neg",
-    template_mode="median",
-    dtype=None,
-    verbose=True,
+    waveform_extractor: WaveformExtractor,
+    output_folder: Union[str, Path],
```
Comment (alejoe91) [inline src/spikeinterface/exporters/to_phy.py:21]: @h-mayorquin @DradeAW would it be better to use:

```
from __future__ import annotations

...
output_folder: str | Path

```

Comment (h-mayorquin) [reply src/spikeinterface/exporters/to_phy.py:21]: I  like `str | Path` better.

**Comment by alejoe91** [comment]
@DradeAW 
```
E       from __future__ import annotations
E       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   SyntaxError: from __future__ imports must occur at the beginning of the file
```

File: src/spikeinterface/exporters/to_phy.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from pathlib import Path
-from typing import Literal, Optional, Union
+from typing import Literal, Optional
+from __future__ import annotations
```
Comment (alejoe91) [inline src/spikeinterface/exporters/to_phy.py:3]: this should be the first line!

Comment (DradeAW) [reply src/spikeinterface/exporters/to_phy.py:3]: Aaah! Didn't know that ^^

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**